vkontakte/content/contents/code/vkontakte.js

```diff
@@ -1,398 +1,339 @@
-/* === This file is part of Tomahawk Player - <http://tomahawk-player.org> ===
+/* vk.com resolver for Tomahawk.
  *
- *   Copyright 2011, Krzysztof Klinikowski <kkszysiu@gmail.com>
+ * Written in 2015 by Anton Romanov
+ * API documentation: https://vk.com/dev
  *
- *   Tomahawk is free software: you can redistribute it and/or modify
- *   it under the terms of the GNU General Public License as published by
- *   the Free Software Foundation, either version 3 of the License, or
- *   (at your option) any later version.
+ * To the extent possible under law, the author(s) have dedicated all
+ * copyright and related and neighboring rights to this software to
+ * the public domain worldwide. This software is distributed without
+ * any warranty.
  *
- *   Tomahawk is distributed in the hope that it will be useful,
- *   but WITHOUT ANY WARRANTY; without even the implied warranty of
- *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- *   GNU General Public License for more details.
- *
- *   You should have received a copy of the GNU General Public License
- *   along with Tomahawk. If not, see <http://www.gnu.org/licenses/>.
+ * You should have received a copy of the CC0 Public Domain Dedication
+ * along with this software. If not, see:
+ * http://creativecommons.org/publicdomain/zero/1.0/
  */
 
-debugMode = false;
+var VkontakteResolver = Tomahawk.extend( Tomahawk.Resolver.Promise, {
+    apiVersion: 0.9,
 
-function xhrRequest(url, method, data, callback){
-    var xhr = new XMLHttpRequest()
+    //These are copied from kodi vk.com plugin
+    APP_ID : '2054573',
+    APP_KEY : 'KUPNPTTQGApLFVOVgqdx',
+    APP_SCOPE : 'audio',
+    API_VERSION : '5.0',
 
-    if (debugMode == true) { Tomahawk.log('Sending request:' + url + '?' + data); }
+    STORAGE_KEY : 'vk.com.access_token',
 
-    if(method == "POST"){
-        xhr.open(method, url, true)
-        xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded")
-        xhr.send(data)
-    } else {
-        if(url.match(/\?/))
-            xhr.open(method, url+'&'+data, true)
-        else
-            xhr.open(method, url+'?'+data, true)
+    logged_in: null, // null, = not yet tried, 0 = pending, 1 = success, 2 = failed
 
-        xhr.send()
-    }
+    _queue : {}, //we'll queue up resolve requests and execute them in batches
```

Comment:
You are later on looping over this with for in, then you need to either remember to check hasOwnProperty on every iteration, or you need to initialize this with `Object.create(null)`